doc/api/modules.md

```diff
@@ -888,12 +888,29 @@ by the [module wrapper][]. To access it, require the `Module` module:
 const builtin = require('module').builtinModules;
 ```
 
+### module.createRequireFunction(filename)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `filename` {string|URL} Filename for relative importing
+* Returns: {[`require`][]} Require function
+
+<!-- eslint-skip -->
+```js
+// create a require function which will resolve relative
+// to the current ES Module file
+import Module from 'module';
+const require = Module.createRequireFunction(new URL(import.meta.url));
+```
+
```

Comment:
Some explanation about *why* this is useful would probably be a good thing in here. It will be non-obvious to many users and the lack of explanation could lead to confusion.

Reply:
Can you also add an example using cjs and require? This is useful outside of ESM.